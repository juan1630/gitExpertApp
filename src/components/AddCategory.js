import React, { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

// se debe de dejar elstrig vacio
  const [inputValue, setInputValue] =useState('');

  const handleValue = (e) => {

      setInputValue( e.target.value );
  }

  const handleSubmit = (e) => {
    // console.log(e );
    e.preventDefault();

    if( inputValue.trim().length > 2 ){
            // se previene que se recargue la pagina
      setCategories( cats => [inputValue, ...cats] );
      // recibimso por defecto los valores de la catgoria
      setInputValue('');
      }


  }

  return(

    // el fargment se quita si hay otro agrupador
    <form   onSubmit={ handleSubmit }  >

      <input type="text"
      className="input"
      value={inputValue}
      onChange={  handleValue }
      />
      </form>
  )
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
