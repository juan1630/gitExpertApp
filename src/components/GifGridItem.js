import React from 'react';


export const GifGridItem = ( {id, url, title} ) => {

        // console.log( props );
        return ( <div className="card animate__animated animate__bounce animate__delay-5s"  >
          <img src={url} alt={title} />
          <p> { title } </p>
         </div>)
        }
