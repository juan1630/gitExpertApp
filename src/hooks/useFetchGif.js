// los use demuestran que es un hook
// los hooks son funciones
//


import  {getsGifs}  from '../helpers/getGifs'
import { useState, useEffect } from 'react';

// el hook si puede tener el estado
export const useFetchGif = ( category ) => {

  const [state, setState] = useState({
    data:[],
    loading:true
  });


  useEffect( ()=> {
        getsGifs(category)
        .then(  imgs => {


          setState({
              data: imgs,
              loading:false
          });

        })
  }, [ category ]);

  return state;
}
