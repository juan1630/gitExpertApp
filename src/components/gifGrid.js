import React from 'react';
import { useFetchGif } from '../hooks/useFetchGif';
import {GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {


      const { data: images, loading }  =  useFetchGif( category );
      // console.log( data );
      // console.log( loading );

      { loading && <p>loading </p> }

        return (
          <div className="" >
            <h3> { category } </h3>
                <div className="card-grid" >
                  {
                    images.map( (img) =>(

                      <GifGridItem key={img.id} {...img} />
                    )
                    )
                  }
                </div>
            </div>)
        }

        // { loading ? 'Cargando...' : 'Fin de la carga' }
