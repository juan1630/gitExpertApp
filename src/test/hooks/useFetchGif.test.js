import { GifGrid } from '../../components/gifGrid';
import { shallow} from 'enzyme';
import React from 'react';

import { useFetchGif } from '../../hooks/useFetchGif';
jest.mock('../../hooks/useFetchGif')


describe('Pruebas del useFetchGif', () => {


  let wrapper;
  const category = "Zelda";

  test('Debe de hacer match con el snapshot', () => {

    // retorna la infomacio del componente
      useFetchGif.mockReturnValue({
        data:[],
        loading:true
      });

      wrapper = shallow(<GifGrid  category={category} /> );
        expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de mostart items cuando se carguen las imagenes', () => {
        // mock fingue que se trajo la data del servicio


        const gifs = [{
          id:'ABC',
          url:'https://localhost:3200/images/photo.jpg',
          title:'algo jeje'
        }]

        useFetchGif.mockReturnValue({
          data:gifs,
          loading:false
        });



        wrapper = shallow(<GifGrid  category={category} /> );


        expect( wrapper ).toMatchSnapshot();
  });
});
