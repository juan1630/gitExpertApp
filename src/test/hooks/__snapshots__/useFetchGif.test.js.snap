// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del useFetchGif Debe de hacer match con el snapshot 1`] = `
<div
  className=""
>
  <h3>
     
    Zelda
     
  </h3>
  <div
    className="card-grid"
  />
</div>
`;

exports[`Pruebas del useFetchGif Debe de mostart items cuando se carguen las imagenes 1`] = `
<div
  className=""
>
  <h3>
     
    Zelda
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="algo jeje"
      url="https://localhost:3200/images/photo.jpg"
    />
  </div>
</div>
`;
